src/main/createWindow.js

```diff
@@ -69,11 +69,22 @@ const createWindow = (pathname, options = {}) => {
       app.addRecentDocument(pathname)
       const filename = path.basename(pathname)
       fs.readFile(path.resolve(pathname), 'utf-8', (err, file) => {
-        if (err) return console.log(err)
+        if (err) {
+          console.log(err)
+          return
+        }
+
+        // check UTF-8 BOM (EF BB BF) encoding
+        let isUtf8BomEncoded = file.length >= 1 && file.charCodeAt(0) === 0xFEFF
```

Comment:
I think you can use a js `Set` to store weather it is UTF BOM encode here?

```javascript
const fileEncodeSet = new Set()
// other code
if (isUtf8BomEncoded) fileEncodeSet.add(pathname)
// other code
if (fileEncodeSet.has(pathname)) {
  // add `0xFEFF` back to content
}
```

In this way, you don't have to save it in the render process(state)?

Reply:
Are you sure you want to do it that way? We have to implement a Set with counter, because you can open the same file multiple times.

Reply:
I look into the codes and still a little confused.

>We have to implement a Set with counter, because you can open the same file multiple times.

In my opinions, if a file is UTF BOM encoded is it always UTF BOM encoded? even if it is opened multiple times, is the `UTF BOM encoded` will be changed? if not, why need to `implement a Set with counter`?

>Then we add files on load/open and remove them on window close or if user opens another file and counter equals 0.

when we load/open a file, we add it pathname to the `Set`, how is it `counter equals 0`?

❓

Reply:
> if not, why need to implement a Set with counter?

See below or user opens file 1 then he opens file 1 again. He closes one window and now we think file hasn't BOM.

If we only add files the Set, it can hold invalid encodings. If we add file and remove it on first window close, another window with same file don't know about encoding.

> when we load/open a file, we add it pathname to the Set, how is it counter equals 0?

We have to remove file from set if conter equals 0 and decrease counter if user closes the window because file BOM can be change. E.g. user deletes the file etc.